okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -106,10 +106,16 @@ public HttpCodec newStream(
     boolean connectionRetryEnabled = client.retryOnConnectionFailure();
 
     try {
+      RealConnection existingConnection = connection;
+
       RealConnection resultConnection = findHealthyConnection(connectTimeout, readTimeout,
           writeTimeout, connectionRetryEnabled, doExtensiveHealthChecks);
       HttpCodec resultCodec = resultConnection.newCodec(client, chain, this);
 
+      if (existingConnection != connection) {
```

Comment:
I'm happy to leave this at it's awkward but correct, so deal with it.  But just giving the feedback that it makes avoiding logging ONLY one of either new connection or reused awkward.